src/witness/witnessConstraints.ml

```diff
@@ -149,6 +149,19 @@ struct
     let d = Dom.fold_keys h (fst ctx.local) (Dom.empty (), Sync.bot ()) in
     if Dom.is_bot (fst d) then raise Deadcode else d
 
+  (* TODO???? *)
+  let map_event ctx e =
+    (* we now use Sync for every tf such that threadspawn after tf could look up state before tf *)
+    let h x (xs, sync) =
+      try
+        let x' = Spec.event (conv ctx x) e (conv ctx x) in
+        (Dom.add x' (step_ctx_edge ctx x) xs, Sync.add x' (SyncSet.singleton x) sync)
+      with Deadcode -> (xs, sync)
+    in
+    let d = Dom.fold_keys h (fst ctx.local) (Dom.empty (), Sync.bot ()) in
+    if Dom.is_bot (fst d) then raise Deadcode else d
```

Comment:
I'll have to think about this sometime later. The meaning of events coming from outside the analysis is a wobbly concept, which is why `event` wasn't part of the `Spec` signature to begin with.